chore(build): Make push-api fail if no usb host was found

sfoster1 opened the pull request:
Previously, because the robot finding mechanism was not actually in a target, if no robot was found
in push-api then the makefile would proceed to build the wheel and only fail when it tried to
connect to :31950 (note the lack of actual IP address there). By instead invoking the discovery
client directly in a target and using a file intermediate we can fail fast if no robot is found.

## Review requests

Check it on windows just to be sure.

mcous commented on Makefile:
```diff
@@ -24,9 +24,10 @@ updateSnapshot ?= false
 	cover := false
 endif
-
 # run at usage (=), not on makefile parse (:=)
```
This comment is no longer applicable

mcous commented on Makefile:
```diff
@@ -24,9 +24,10 @@ updateSnapshot ?= false
-usb_host = $(shell yarn run -s discovery find -i 169.254 fd00 -c "[fd00:0:cafe:fefe::1]")
+.PHONY: usb-connect
+usb-connect:
+	yarn run -s discovery find -i 169.254 fd00 -c "[fd00:0:cafe:fefe::1]" >.usb_host
```
`yarn run -s` is not necessary here, because inside a task `discovery` is in the `PATH` (it is not in the path for anything run inside `$(shell ...)`

mcous commented on Makefile:
```diff
@@ -60,7 +61,8 @@ install-types:
 .PHONY: push-api
-push-api: export host = $(usb_host)
+push-api: usb-connect
+push-api: export host=$(shell cat .usb_host)
```
`cat` is not available on Windows, so this will break `make push` on Windows. I think `$(shell yarn run -s shx cat .usb_host)` will work

sfoster1 commented:
Addressed comments made by @mcous (mostly by using the snippets he slacked me). Both push-api and term now
- correctly connect to a usb machine
- correctly fail fast with a message if no usb machine is connected and no host is specified
- connect to an arbitrary host if specified, regardless of whether a usb machine is connected
